File: AllSpice/Models/Instruction.cs
namespace AllSpice.Models;

public class Instruction : ICreated, IRepoItem<int>
{
  public int Id { get; set; }
  public int RecipeId { get; set; }

  public string Step { get; set; }

  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}

File: AllSpice.client/src/models/Instruction.js
import { Account } from "./Account.js";

export class Instruction {
  constructor(data) {
    this.id = data.id;
    this.steps = data.steps;
    this.recipeId = data.recipeId;
    this.creator = new Account(data.creator) || null;
  }
}
